cmd/query-tee/main.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"flag"
+	"os"
+	"time"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/weaveworks/common/logging"
+	"github.com/weaveworks/common/server"
+
+	"github.com/cortexproject/cortex/pkg/util"
+)
+
+type Config struct {
+	ServerServicePort  int
+	ServerMetricsPort  int
+	BackendEndpoints   string
+	PreferredBackend   string
+	BackendReadTimeout time.Duration
+	LogLevel           logging.Level
+}
+
+func main() {
+	// Parse CLI flags.
+	cfg := Config{}
+	flag.IntVar(&cfg.ServerServicePort, "server.service-port", 80, "The port where the query-tee service listens to.")
+	flag.IntVar(&cfg.ServerMetricsPort, "server.metrics-port", 9900, "The port where metrics are exposed.")
```

Comment:
Why two servers? Main server can map some subtree (`/api/*` to backend endpoints, and keep the rest, including `/metrics` for itself).

Reply:
This tool can be publicly exposed given it's the very entrypoint of the system (in front of any "cortex auth gateway"). For this reason, we may want to expose `/metrics` only privately while the read API endpoint will be public.

Reply:
Makes sense, thanks for explanation.